crates/next-core/js/src/internal/utils.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Converts an array of raw header entries to a map of header names to values.
+ */
+export function headersFromEntries(
```

Comment:
Is there a reason this couldn't have been done on the Rust side? I've never like the bytes/string code in our current `Headers` impl. If possible, I'd love to remove that, and just have a single headers instead of `headers` and `raw_headers`.

Reply:
I think this could be moved to Rust without any issues. I'll look into it.

We still need `headers` and `raw_headers` because our `NeedData` implementation is able to filter on header keys and values in order to cache some results.

We do however have quite a few header types across the codebase, so it would be helpful to standardize on only two or three.